src/LlmTornado.Agents/Utility/JsonUtility.cs

```diff
@@ -128,6 +131,86 @@ public static string MapClrTypeToJsonType(Type type)
         if (type.IsArray) return "array";
         return "object"; // fallback for complex types
     }
+
+    /// <summary>
+    /// Attempts to parse a JSON string into an object of type <typeparamref name="T"/>.  If the input is invalid JSON,
+    /// the method attempts to repair it using an external agent.
+    /// </summary>
+    /// <remarks>This method first attempts to parse the input JSON string directly. If the input is invalid
+    /// JSON,  it uses the provided <see cref="TornadoAgent"/> to attempt to repair the JSON and then retries parsing.
+    /// The method performs basic cleanup on the input string, such as removing Markdown code fences and trimming
+    /// whitespace.</remarks>
+    /// <typeparam name="T">The type to which the JSON string should be deserialized.</typeparam>
+    /// <param name="agent">An instance of <see cref="TornadoAgent"/> used to repair invalid JSON if the initial parsing fails. (Can be any agent will restore after use)</param>
+    /// <param name="possibleJson">A string that is expected to contain JSON data. The string may include extraneous formatting, such as Markdown
+    /// code fences.</param>
+    /// <returns>An object of type <typeparamref name="T"/> if the JSON is successfully parsed or repaired;  otherwise, <see
+    /// langword="null"/> if parsing and repair attempts fail.</returns>
+    public static async Task<T?> SmartParseJsonAsync<T>(TornadoAgent agent, string possibleJson)
```

Comment:
Generally a cool idea, might consider moving this up in the library - when are we using this? I'd hope we use as much structured outputs as possible.